13_07_2017/3D_slab_modes.py

```diff
@@ -0,0 +1,779 @@
+    def disp_rel_asym_2var(W, K):
+        return sf.disp_rel_asym(W, K, R1)
+    
+    # find eigenfrequencies W (= omega/k) within the range Wrange for the given parameters.
+    Wrange1 = np.linspace(0., sf.cT, 11)
+    Wrange2 = np.linspace(sf.cT, sf.c0, 401)
+    Wrange3 = np.linspace(sf.c0, sf.c2, 11)
+    
+    Woptions_slow_surf = np.real(tool.point_find(disp_rel_asym_2var, np.array(K), Wrange1, args=None).transpose())
+    Woptions_slow_body = np.real(tool.point_find(disp_rel_asym_2var, np.array(K), Wrange2, args=None).transpose())
+    Woptions_fast = np.real(tool.point_find(disp_rel_asym_2var, np.array(K), Wrange3, args=None).transpose())
+    
+    # Remove W values that are very close to characteristic speeds - these are spurious solutions
+    tol = 1e-2
+    indices_to_rm = []
+    for i in range(len(Woptions_slow_surf)):
+        w = Woptions_slow_surf[i]
+        if min(abs(np.array([w, w - sf.c0, w - sf.c1(R1), w - sf.c2, w - sf.vA]))) < tol or w < 0 or w > sf.cT:
+            indices_to_rm.append(i)
+    Woptions_slow_surf = np.delete(Woptions_slow_surf, indices_to_rm)
+    Woptions_slow_surf.sort()
+    
+    indices_to_rm = []
+    for i in range(len(Woptions_slow_body)):
+        w = Woptions_slow_body[i]
+        if min(abs(np.array([w, w - sf.c0, w - sf.c1(R1), w - sf.c2, w - sf.vA]))) < tol or w < sf.cT or w > sf.c0:
+            indices_to_rm.append(i)
+    Woptions_slow_body = np.delete(Woptions_slow_body, indices_to_rm)
+    Woptions_slow_body.sort()
+    
+    indices_to_rm = []
+    for i in range(len(Woptions_fast)):
+        w = Woptions_fast[i]
+        if min(abs(np.array([w, w - sf.c0, w - sf.c1(R1), w - sf.c2, w - sf.vA]))) < tol or w < sf.c0 or w > min(sf.c1, sf.c2):
+            indices_to_rm.append(i)
+    Woptions_fast = np.delete(Woptions_fast, indices_to_rm)
+    Woptions_fast.sort()
+    
+    # remove any higher order slow body modes - we only want to do the first 3 saus/kink
+    if len(Woptions_slow_body) > 6:
+        Woptions_slow_body = np.delete(Woptions_slow_body, range(len(Woptions_slow_body) - 6))
+    Woptions = np.concatenate((Woptions_slow_surf, Woptions_slow_body, Woptions_fast))
+    
+    # set W to be the eigenfrequency for the requested mode
+    if mode in ['fast-saus-body-1', 'fast-saus-body-2', 'fast-saus-body-3', 'fast-kink-surf']:
+        W = Woptions_fast[-2]
+    elif mode in ['fast-kink-body-1', 'fast-kink-body-2', 'fast-kink-body-3', 'fast-saus-surf']:
+        W = Woptions_fast[-1]
+    elif mode in slow_surf_mode_options:
+        W = Woptions_slow_surf[mode_ind]
+    elif mode in slow_body_mode_options:
+        W = Woptions_slow_body[mode_ind-2]
+    
+    if mode in alfven_mode_options:
+        W = sf.vA
+    else:
+        W = np.real(W)
+        
+    # Quick plot to see if we are hitting correct mode    
+    if show_quick_plot == True:
+        plt.plot([K] * len(Woptions), Woptions, '.')
+        plt.plot(K+0.5, W, 'go')
```

Comment:
Nothing wrong with this at all, but personally I prefer using `color='green', marker='circle'` (or whatever) in `plt.plot()` rather than the colour/marker string thingy, I find it much clearer. Just saying in case you weren't aware of it.